\chapter{Abbreviations and Notation}
\label{ch:notation}

%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Notation
%% --------
%%
%% n: number of probes
%% m: number of spots
%% t: step index
%% N: deposition sequence
%% N_t: nucleotide added at step t
%% M_t: mask of step t
%% T: total number of synthesis steps
%%
%% s,s': spot indices
%% p_k: probe sequence of probe with index k
%% k(s): probe index for spot s
%% \eps_k: embedding vector for probe with index k
%% \eps_{k(s)}: embedding vector at spot s
%%
%% \mathcal{B}_t: border length of mask M_t
%%
%% For the QAP formulation (not here!)
%% k,l: indices for probes
%% i,j: indices for spots
%%
%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% See also
%% http://www.mcponline.org/glossary/
%%
\small
\vspace*{-4ex}
\subsection*{Abbreviations in alphabetical order}
\begin{longtable}[l]{rl}
\endfirsthead\endhead\endfoot\endlastfoot
%%
%% A
A, \tA    & adenine\\
\Angstrom   & Angstrom; $1\Angstrom = 10^{-10}\text{ m}=1/10 \text{ nm}$\\
aRNA        & anti-sense RNA, amplified RNA; complementary to mRNA\\
%%
%% B
bp          & base pair(s)\\
%%
%% C
C, \tC    & cytosine\\
\degC       & degrees centigrade (Celsius)\\
cDNA        & complementary DNA; DNA synthesized from mRNA by RT\\
CDS         & coding sequence\\
cRNA        & complementary RNA; same as aRNA\\
CSMS, csms  & cumulative statistics of matching statistics\\
Cy3         & Cyanine 3-dNTP; in fluorescently labeled DNA\\
Cy5         & Cyanine 5-dNTP; in fluorescently labeled DNA\\
%%
%% D
DMD         & digital micromirror device\\
DNA         & deoxyribonucleic acid\\
dN          & any deoxynucleotide; any of dA, dC, dG, dT\\
dNTP        & any deoxynucleotide-triphosphate; 
                any of dATP, dCTP, dGTP, dTTP\\
%%
%% E
EST         & expressed sequence tag\\ 
%%
%% G
G, \tG    & guanine\\
%%
%% H
HPLC        &  high-pressure liquid chromatography\\
%%
%% I
IVT         & in-vitro transcription\\
%% 
%% K
K           & Kelvin; physical unit of temperature\\
%%
%% L
LCF         & longest common factor\\
LCFS        & longest common factor statistics\\
%%
%% M
M           &  physical unit of molar concentration: 1 M = 1 mol/l\\
MCMC        &  Markov chain Monte Carlo\\
MES         &  2-(N-morpholino)ethanesulfonic acid\\
mol         &  physical unit of amount of substance\\
{}          &{\qquad}1 mol = as many entities as atoms in $0.012$ kg of carbon 12\\
mRNA        &  messenger RNA\\
MS, ms      &  matching statistics\\
%%
%% N
N, \tN    &  any of the bases A, C, G, T/U\\
{[\Naplus]} &  salt (sodium ion) concentration\\
NaCl        &  salt (sodium chloride)\\
NN          &  nearest neighbor\\
nt          &  nucleotide(s)\\
%%
%% O
oligo-(dT)  &  primer of 12--20 dTs binding to the poly(A) tail of 
                 eukaryotic mRNA\\
%%
%% P
P           &  phosphor\\
${}^{33}$P  &  a radioactive phosphor isotope\\
PAGE        &  polyacrylamide gel electrophoresis\\
PCR         &  polymerase chain reaction\\
PEM         &  percent of expected mutations (unit of evolutionary time)\\
%%
%% R
RNA         & ribonucleic acid\\
RT          & reverse transcription\\
RTase       & reverse transcriptase; an enzyme\\
RT-PCR      & reverse transcription-polymerase chain reaction\\
%%
%% S
SAGE        & serial analysis of gene expression\\
SAPE        & streptavidin phycoerythrin\\
SBH         & sequencing by hybridization\\
SCS         & shortest common supersequence\\
SCSP        & shortest common supersequence problem\\
SNP         & single nucleotide polymorphism\\
SSC         & saline sodium citrate buffer\\
SSPE        & saline sodium phosphate buffer\\
SVD         & singular value decomposition\\
%%----------------------------------------------
%% T
T, \tT    &  thymine\\
T7          &  a bacteriophage that infects \emph{E.\ coli};\\
{}          &{\qquad}these viral parasites are numbered T1 through T7\\
\textit{Taq}&  \textit{Thermus aquaticus}; a heat-resistant bacterium\\
TIFF        &  tagged image file format\\
transfrag   &  transcript fragment\\
tRNA        &  transfer RNA\\
%%----------------------------------------------
%% U 
U, \tU    &  uracil\\
UTR         &  untranslated region; part of mRNA transcripts\\
%%
%%
\end{longtable}


\subsection*{Notation by topic}
\newcommand{\ltchap}[1]{\multicolumn{2}{l}{\sffamily\textbf{Chapter~{#1}}}\\\nopagebreak}

Notation introduced in early chapters is also used in subsequent chapters.
\begin{longtable}[l]{rl}
\endfirsthead\endhead\endfoot\endlastfoot
%%
%% Chapter 1
\ltchap{1}
$p$      & probe sequence; a string\\
$|p|$    & length of $p$\\
$t$      & transcript or target sequence\\
$T$      & transcriptome $T=(t_1,\dots,t_n)$\\
$m$      & number of probes\\
$i$      & probe index\\
$n$      & number of transcripts\\
$j$      & transcript index\\
$\theta$ & hybridization conditions; parameters\\
$a(p,t;\theta)$ & affinity coefficient between $p$ and $t$, given $\theta$\\
$p\substring t$ & $p$ is a factor (substring) of $t$\\
$\eps$   & lower affinity limit for specific hybridization\\[2ex]
%%
%% Chapter 2
\ltchap{2}
$x=(x_j)$     & $n$-vector of transcript expression levels\\
$y=(y_i)$     & $m$-vector of measured probe signals\\
$A=(A_{ij})$  & $m\times n$-matrix of probe-transcript affinity coefficients; $y=A\cdot x$\\
$t(i)$        & index of the unique target that probe $i$ binds to\\
$a$, $a_i$    & particular affinity coefficients\\
$\rho_{ij}$   & position dependent factor of $A_{ij}$\\
$\beta_{ij}$  & hybridization stability dependent factor of $A_{ij}$\\
$\gamma_{ij}$ & sequence composition dependent factor of $A_{ij}$\\
$\sigma_{ij}$ & self-complementarity dependent factor of $A_{ij}$\\
$\eps_i$      & stochastic additive noise of probe signal intensity\\
$c_i$, $c$    & systematic additive offset of signal intensity\\
$\alpha$      & scale of signal intensity\\
$\eta_{ij}$   & stochastic multiplicative noise of signal intensity\\[1ex]
$\diff$       & differential operator\\
$\Delta$      & denotes a difference\\
$U$; $q$; $w$ & internal energy; heat; work\\
$p$           & pressure\\
$V$           & volume\\
$T$           & temperature\\
$H$; $\stdm{H}$; $\Drstd{H}$ & enthalpy; standard molar enthalpy; standard reaction enthalpy\\
$S$; $\stdm{S}$; $\Drstd{S}$ & entropy; standard molar entropy; standard reaction entropy\\
$G$; $\stdm{G}$              & Gibbs energy; standard molar Gibbs energy\\
$\Drstd{G}$; $\Dr{G}$        & standard Gibbs energy of reaction; reaction Gibbs energy\\
$\xi$         & extent of reaction\\
$R$           & gas constant; $R = 8.3145 \text{ J K$^{-1}$ mol$^{-1}$}$\\
$K$           & equilibrium constant\\
$\conceq{\text{S}_2}$ & equilibrium concentration of reactant $\text{S}_2$ (single stranded probes)\\
%%$\beta$       & hybridization probability, stability factor\\
$\TM$         & melting temperature\\[2ex]
%%
%% Chapter 3
\ltchap{3}
$\lcf(p,t)$      & length of the longest common factor of $p$ and $t$\\
$\lcfe(p,t)$     & same, allowing one mismatch\\
$\lcf^*(p,t)$    & combined measure derived from $\lcf$ and $\lcfe$\\
$\lcfvec(p\given T)$  & LCF vector of probe $p$ against transcriptome $T$\\
$\lcfstat(p\given T;\Delta)$  & LCF statistics of $p$ against $T$ of width $\Delta$\\
$\delta$         & index of LCF statistics, $0\leq\delta<\Delta$\\
& \atab difference between full probe length and LCF length\\
$T(i)$           & index set of intended targets for probe $i$\\
$T$              & transcriptome; see Chapter~1\\
$U'(p_i\given T)$& unspecificity of probe $i$ in $T$; formal definition\\
$\tau$           & temperature; to avoid confusion with transcriptome $T$\\
$\beta(\delta)$  & average hybridization probability as a function of $\delta$\\
$\zeta$          & offset constant; $\zeta=\ln(\beta(0)/(1-\beta(0)))$\\
$b>0$            & average Gibbs energy per base pair in units of $(R\tau)$\\
$u(\delta)$      & approximation to $\beta(\delta)$\\
$U(p_i\given T)$ & unspecificity of probe $i$ in $T$; practical definition\\[2ex]
%%
%% Chapter 4
\ltchap{4}
$T=(T_c)$           & transcriptome of transcript collections $T_c$ ($c=1..C$)\\
$C$                 & number of collections\\
$T_c = \{t_{c,j}\}$ & transcript collection $c$ with transcripts $t_{c,j}$ ($j=1..N_c)$\\
$N_c$               & number of transcripts in collection $c$\\
$s$                 & target sequence\\
$\Sigma$            & alphabet; here $\Sigma=\{\Nucs{A,C,G,T}\}$\\
$\Sigma^*$; $\Sigma^+$ & all strings (non-empty strings) over alphabet $\Sigma$\\
$\sep$              & string end marker and separator\\
$\tX$               & unspecified or wildcard character\\
$\pos$              & suffix array\\
$\lcpr(s,t)$        & length of the longest common prefix of $s$ and $t$\\
$\lcp$              & longest common prefix array\\
$q$                 & bucket prefix length\\
$\bck$              & bucket array\\
$\gamma=\qcode{Q}$  & numerical radix-$q$ representation of a $q$-gram $Q$\\
$\cl$               & collection number array\\
$\ms^{s|t}=(\ms^{s|t}_i)$ & matching statistics of $s$ against $t$\\
$\ms^{s|t;f}$       & matching statistics allowing $f$ mismatches of $s$ against $t$\\
$\Romin$            & minimum value of relevant matching statistics\\
$\Remin$            & minimum value of relevant ms.\ with one mismatch\\
$\Rmax$             & maximum value of relevant matching statistics\\
$\MS[i][c]$         & matching statistics array; $\MS[i][c]=\ms^{s|T_c}_i$\\[1ex]
%%
$(i,J)$             & jump at position $i$ to level $J$\\
$n,m$               & string lengths; $|s|=m$, $|t|=n$\\
$\Prob$             & generic notation for a probability measure\\
$\Erw$              & generic notation for an expectation\\
$\pi=(\pi_c)_{c\in\Sigma}$ & character distribution for a random sequence model\\
$p$                 & probability that two random characters match\\
$q$                 & $q:=1-p$\\
$K_L$               & random number of occurrences of a random $L$-gram in a string\\
$\rho_L$            & probability that a jump to level $L$ occurs at a fixed position\\
$E_L$               & expected number of jumps to level $L$ in matching statistics\\
$E^+_L$             & expected number of jumps to level $\geq L$ in matching statistics\\
$\Lo$               & center of the distribution of the LCF of two random strings\\[2ex]
%%
%% Chapter 5
\ltchap{5}
$d$      & probe distance from the transcript's $3'$-end\\
$p_i=(d_i,\ell_i)$ & probe $i$ defined by end distance $d_i$ and length $\ell_i$\\
$f(d)$; $f_i$ & position preference factor for distance $d$; $f_i:=f(d_i)$\\
$h_i$    & hairpin formation probability for probe $i$\\
$U_i$    & unspecificity for probe $i$\\
$B_i$    & badness value for probe $i$\\
$\CalB(\delta)$ & additional badness for probes clustering within distance $\delta$\\
$B'_i$; $\Bbar'$ & modified badness value for probe $i$; threshold $\Bbar'$\\
$S$           & selected probe set\\[2ex]
%%
%%
\ltchap{6}
$\Ind{\cdot}$ & generic notation for an indicator function\\
$S=(s_c)$     & target sequences for all collections $c=1,\dots,C$\\
$H=(H_{ij})$  & binary $m\times n$ hybridization matrix of probes vs.\ targets\\
$D$           & design, i.e. a selection of probes; $D\subset\{1,\dots,m\}$\\
$A^D$; $H^D$  & affinity and hybridization matrix restricted to rows from $D$\\
$\mcov$       & minimum target coverage\\
$\acov$       & average target coverage\\
$A\tp$        & transpose of $A$\\
$\Sigma$      & diagonal matrix with singular values $(\sigma_j)$ of $A$\\
$\sigma_j$    & $j$-th largest singular value\\
$\diag(\cdot)$& diagonal matrix with specified entries\\
$A^-$         & pseudo-inverse of $A$\\
$\|\cdot\|_2$ & Euclidean norm of a vector; spectral norm of a matrix\\
$\|\cdot\|_\infty$ & maximum norm of a vector\\
$\cond(A)$    & condition number of $A$\\
$\scond(A)$   & Skeel condition number of $A$\\
$\mu$         & maximal number of probes that may be selected\\[1ex]
%%
$S$,$T$       & sets of target indices\\
$S\Delta T$   & symmetric set difference of $S$ and $T$\\
$P(S)$        & set of probes that hybridize to any target in $S$\\
$T(i)$        & set of targets that hybridize to probe $i$\\
$\gamma$      & desired minimum target coverage\\
$\sigma$      & desired minimum target set separation\\
$\delta\in\{0,1\}^m$ & binary vector representation of design $D$\\
$h_j$         & $j$-th column of binary hybridization matrix $H$\\
$z^{S,T}=(z^{S,T}_i)$  & indicates whether probe $i$ separates sets $S$ and $T$\\
$\delta^+$, $h_j^+$, $z^+$ & vectors extended for virtual unique probes\\
$B$           & maximal size of target sets to be separated\\
$f_0$; $f_1$  & false negative (positive) error rate\\
$\Prob(x)$    & prior probability of binary expression vector $x\in\{0,1\}^n$\\
$\Prob(y\given x)$ & likelihood of probe signals $y\in\{0,1\}^m$, given $x$\\
$\pi(x) = \Prob(x\given y)$ & posterior probability of $x$ for fixed $y$\\
$q(z \given x)$ & proposal probability for new solution $z$ at current solution $x$\\
$\CalN(x)$      & neighborhood of $x$\\
$\alpha(x,z)$   & acceptance probability for $z$ at $x$\\[1ex]
%%
$Q$             & rate matrix of an evolutionary Markov process\\
$t$             & evolutionary time\\
$P^t$           & Markov transition kernel for time $t$\\
$\expm(Q)$      & matrix exponential; $\expm(Q):=\sum_{n\geq 0}\, Q^n/n!$\\[2ex]
%%
%% Chapter 7
\ltchap{7}
$\csms^{s|t}$   & cumulative statistics of matching statistics of $s$ against $t$\\
& \atab $\csms^{s|t}_{i,\mu}$ refers to position $i$ in $s$ and matches of length $\geq \mu$\\
$\CSMS[i][\mu]$ & CSMS array; $\CSMS[i][\mu]=\csms^{s|t}_{i,\mu}$ if $\mu\in[\Romin,\Rmax]$\\
$\sigma^i_{k,u}$& unexpected CSMS at offset $k$ for a probe starting at $i$\\
$L_{i,\delta}$  & whole genome surrogate for $\lcfstat(p_i)_\delta$\\[1ex]
$L$             & length of hypothetical transfrags\\
$D$             & distance between start points of hypothetical transfrags\\
$K$             & transfrags known to be present\\
$P_1$           & probes expected to show a signal because of $K$\\
$M$             & transfrags of unknown status; {$M=\{1..n\} \setminus K$}\\
$N$             & transfrags most likely not present\\
$J$             & transfrags whose status is inferred by sampling; $J=\{1..n\}\setminus(K\cup N)$\\
$\chi$          & fraction of transfrags hybridizing to additional probes\\[2ex]
%%
%% Chapter 8
\ltchap{8}
$\Sigma$    & DNA alphabet\\
$\pi$       & permutation of the nucleotides\\
$\CalS$     & sequence set\\
$e$         & binary embedding vector of a string into a supersequence\\
$E$         & embedding matrix with rows $e_i$, $i=1..|\CalS|$\\
$\CalU$     & upper bound on the supersequence length\\
$\CalL$     & lower bound on the supersequence length\\
$L_i$       & length of the $i$-th sequence in $\CalS$\\
$N_i(x)$    & number of occurrences of $x\in\Sigma$ in the $i$-th sequence\\
$N(x)$      & $\max_i\, N_i(x)$\\
$C_i$       & completion step of the $i$-th sequence\\
$W_{i,k}$   & number of unproductive steps between $(k-1)$-st and $k$-th productive\\&\atab step for sequence $i$\\
$\Phi$      & cumulative distribution function of the standard Normal distribution
\end{longtable}
\normalsize


\paragraph{Trademark Notice.} 
Affymetrix\textR\ and GeneChip\textR\ are registered trademarks of
Affy\-metrix, Inc., Santa Clara, CA, U.S.A.  The terms {\sffamily
  geniom}\textR\ {\sffamily one} and DNA processor\textR\ are
registered trademarks of febit AG, Mannheim, Germany.
